qa/tasks/cephfs/caps_helper.py

```diff
@@ -1,80 +1,195 @@
+                fsname = moncap[fsname_first_char : last]
+                moncap = moncap.replace(moncap[0 : last+1], '')
+            else:
+                fsname = moncap[fsname_first_char : ]
+                moncap = moncap.replace(moncap[0 : ], '')
+
+            fsnames.append(fsname)
+
+        return fsnames
+
+    def run_mon_cap_tests(self, def_fs, client_id):
+        """
+        Check that MON cap is enforced for a client by searching for a Ceph
+        FS name in output of cmd "fs ls" executed with that client's caps.
 
-        fsls = self.run_cluster_cmd(f'fs ls --id {self.client_id} -k '
-                                    f'{keyring_path}')
+        def_fs stands for default FS on Ceph cluster.
+        """
+        get_cluster_cmd_op = def_fs.mon_manager.raw_cluster_cmd
+
+        keyring = get_cluster_cmd_op(args=f'auth get client.{client_id}')
+
+        moncap = None
+        for line in keyring.split('\n'):
+            if 'caps mon' in line:
+                moncap = line[line.find(' = "') + 4 : -1]
+                break
+        else:
+            raise RuntimeError('run_mon_cap_tests(): mon cap not found in '
+                               'keyring. keyring -\n' + keyring)
+
+        keyring_path = def_fs.admin_remote.mktemp(data=keyring)
+
+        fsls = get_cluster_cmd_op(
+            args=f'fs ls --id {client_id} -k {keyring_path}')
+        log.info(f'output of fs ls cmd run by client.{client_id} -\n{fsls}')
 
         if 'fsname=' not in moncap:
-            fsls_admin = self.run_cluster_cmd('fs ls')
+            log.info('no FS name is mentioned in moncap, client has '
+                     'permission to list all files. moncap -\n{moncap}')
+            log.info('testing for presence of all FS names in output of '
+                     '"fs ls" command run by client.')
+
+            fsls_admin = get_cluster_cmd_op(args='fs ls')
+            log.info('output of fs ls cmd run by admin -\n{fsls_admin}')
+
             self.assertEqual(fsls, fsls_admin)
             return
 
-        fss = (self.fs1.name, self.fs2.name) if hasattr(self, 'fs1') else \
-            (self.fs.name,)
-        for fsname in fss:
-                if fsname in moncap:
-                    self.assertIn('name: ' + fsname, fsls)
-                else:
-                    self.assertNotIn('name: ' + fsname, fsls)
+        log.info('FS names are mentioned in moncap. moncap -\n{moncap}')
+        log.info('testing for presence of these FS names in output of '
+                 '"fs ls" command run by client.')
+        for fsname in self._get_fsnames_from_moncap(moncap):
+            self.assertIn('name: ' + fsname, fsls)
 
-    def run_mds_cap_tests(self, filepaths, filedata, mounts, perm):
-        self.conduct_pos_test_for_read_caps(filepaths, filedata, mounts)
+    def run_mds_cap_tests(self, perm, mntpt=None):
+        """
+        Run test for read perm and run write perm, run positive test if it
```

Comment:
I think `it` here means the `write perm`, right ?

Reply:
`and, for write perm, `

Reply:
Fixed now.

Reply:
https://github.com/ceph/ceph/pull/46994/commits/370449ac9d088ba33111fb4948b1bf59f7901168#diff-9998476d9d02f4136e9f6b4b512be93a77a5911a30027bec6a120706a1a024f3R88-R89